### PR TITLE
Update Facebook Events sample to use the latest design patterns

### PR DESCRIPTION
**Pull request by EricDahlvang**

### CODE REVIEW

File: samples/csharp_dotnetcore/23.facebook-events/Dialogs/MainDialog.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        private async Task<DialogTurnResult> ChoiceStepAsync(WaterfallStepContext stepContext, CancellationToken cancellationToken)
+        {
+            Logger.LogInformation("MainDialog.ChoiceStepAsync");
+            var turnContext = stepContext.Context;
+
+            var text = turnContext.Activity.Text;
+            // If the text is one of the quick reply options, continue to the ShowResultStep
+            if (new[] { QuickRepliesOption, FacebookPageNameOption, PostBackOption }.Contains(text))
+            {
+                return await stepContext.NextAsync();
```
Comment (gabog) [inline samples/csharp_dotnetcore/23.facebook-events/Dialogs/MainDialog.cs]: Pass cancellationToken() on NextAsync()

File: samples/csharp_dotnetcore/23.facebook-events/Dialogs/MainDialog.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            // Initially the bot offers to showcase 3 Facebook features: Quick replies, PostBack and getting the Facebook Page Name.
+            switch (turnContext.Activity.Text)
+            {
+                // Here we showcase how to obtain the Facebook page name.
+                // This can be useful for the Facebook multi-page support provided by the Bot Framework.
+                // The Facebook page name from which the message comes from is in turnContext.Activity.Recipient.Name.
+                case FacebookPageNameOption:
+                    {
+                        var reply = turnContext.Activity.CreateReply($"This message comes from the following Facebook Page: {turnContext.Activity.Recipient.Name}");
+                        await turnContext.SendActivityAsync(reply);
```
Comment (gabog) [inline samples/csharp_dotnetcore/23.facebook-events/Dialogs/MainDialog.cs]: Pass cancellationToken to SendActivityAsync()

File: samples/csharp_dotnetcore/23.facebook-events/Dialogs/MainDialog.cs [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+            // Initially the bot offers to showcase 3 Facebook features: Quick replies, PostBack and getting the Facebook Page Name.
+            switch (turnContext.Activity.Text)
+            {
+                // Here we showcase how to obtain the Facebook page name.
+                // This can be useful for the Facebook multi-page support provided by the Bot Framework.
+                // The Facebook page name from which the message comes from is in turnContext.Activity.Recipient.Name.
+                case FacebookPageNameOption:
+                    {
+                        var reply = turnContext.Activity.CreateReply($"This message comes from the following Facebook Page: {turnContext.Activity.Recipient.Name}");
```
Comment (gabog) [inline samples/csharp_dotnetcore/23.facebook-events/Dialogs/MainDialog.cs]: Not sure that this is correct, turnContext.Activity.Recipient.Name is displaying my bot handle as registered in Bot Service, it is not showing my Facebook page name.

File: samples/csharp_dotnetcore/23.facebook-events/Dialogs/MainDialog.cs [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+                            Buttons = new List<CardAction>
+                                    {
+                                        new CardAction() { Title = "Yes", Type = ActionTypes.PostBack, Value = "Yes" },
+                                        new CardAction() { Title = "No", Type = ActionTypes.PostBack, Value = "No" },
+                                    },
+                        };
+
+                        var reply = turnContext.Activity.CreateReply();
+                        reply.Attachments = new List<Attachment> { card.ToAttachment() };
+                        await turnContext.SendActivityAsync(reply);
```
Comment (gabog) [inline samples/csharp_dotnetcore/23.facebook-events/Dialogs/MainDialog.cs]: add cancellationToken to SendActivityAsync() call.

File: samples/csharp_dotnetcore/23.facebook-events/Dialogs/MainDialog.cs [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+                // By default we offer the users different actions that the bot supports, through quick replies.
+                case QuickRepliesOption:
+                default:
+                    {
+                        await ShowChoices(stepContext, cancellationToken);
+
+                        break;
+                    }
+            }
+            return await stepContext.EndDialogAsync();
```
Comment (gabog) [inline samples/csharp_dotnetcore/23.facebook-events/Dialogs/MainDialog.cs]: pass cancellationToken to EndDialogAsync call:
    return await stepContext.EndDialogAsync(cancellationToken: cancellationToken);

File: samples/csharp_dotnetcore/23.facebook-events/Bots/FacebookBot.cs [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+            Logger.LogInformation("Running dialog with Message Activity.");
+
+            // Run the Dialog with the new message Activity.
+            await Dialog.Run(turnContext, ConversationState.CreateProperty<DialogState>("DialogState"), cancellationToken);
+        }
+
+        protected override Task OnEventActivityAsync(ITurnContext<IEventActivity> turnContext, CancellationToken cancellationToken)
+        {
+            // Analyze Facebook payload from channel data.
+            ProcessFacebookPayload(turnContext.Activity.ChannelData);
```
Comment (gabog) [inline samples/csharp_dotnetcore/23.facebook-events/Bots/FacebookBot.cs]: I tied debugging the bot against FB Messenger directly and this override is not getting called, am I missing some config steps on my bot?

File: samples/csharp_dotnetcore/23.facebook-events/FacebookModel/FacebookMessage.cs [hunk 7]
```diff
@@ -3,7 +3,7 @@
 
 using Newtonsoft.Json;
 
-namespace Facebook_Events_Bot.FacebookModel
+namespace Microsoft.BotBuilderSamples.FacebookModel
 {
```
Comment (gabog) [inline samples/csharp_dotnetcore/23.facebook-events/FacebookModel/FacebookMessage.cs]: Missing cref in IsEcho property below, consider adding a proper link or removing the cref from /// comments.

File: samples/csharp_dotnetcore/23.facebook-events/Dialogs/MainDialog.cs (hunk 1 above)
Comment (EricDahlvang) [reply samples/csharp_dotnetcore/23.facebook-events/Dialogs/MainDialog.cs]: Refactored, removing the main dialog

File: samples/csharp_dotnetcore/23.facebook-events/Dialogs/MainDialog.cs (hunk 2 above)
Comment (EricDahlvang) [reply samples/csharp_dotnetcore/23.facebook-events/Dialogs/MainDialog.cs]: addressed

File: samples/csharp_dotnetcore/23.facebook-events/Dialogs/MainDialog.cs (hunk 3 above)
Comment (EricDahlvang) [reply samples/csharp_dotnetcore/23.facebook-events/Dialogs/MainDialog.cs]: Changed to Facebook Page Id

File: samples/csharp_dotnetcore/23.facebook-events/Dialogs/MainDialog.cs (hunk 5 above)
Comment (EricDahlvang) [reply samples/csharp_dotnetcore/23.facebook-events/Dialogs/MainDialog.cs]: Refactored, removed MainDailog

File: samples/csharp_dotnetcore/23.facebook-events/FacebookModel/FacebookMessage.cs (hunk 7 above)
Comment (EricDahlvang) [reply samples/csharp_dotnetcore/23.facebook-events/FacebookModel/FacebookMessage.cs]: added link to facebook docs, thanks!

File: samples/csharp_dotnetcore/23.facebook-events/Dialogs/MainDialog.cs (hunk 4 above)
Comment (EricDahlvang) [reply samples/csharp_dotnetcore/23.facebook-events/Dialogs/MainDialog.cs]: addressed

File: samples/csharp_dotnetcore/23.facebook-events/Bots/FacebookBot.cs (hunk 6 above)
Comment (EricDahlvang) [reply samples/csharp_dotnetcore/23.facebook-events/Bots/FacebookBot.cs]: Refactored the sample after discussion with @Kaiqb

**Comment by EricDahlvang** [comment]
After discussing with @Kaiqb I've removed the dialogs and state from this sample, since it is not needed.  The focus is on a simplified processing of Facebook specific payloads and messages.

Feedback has also been addressed.  Thanks for your review @gabog !

File: samples/csharp_dotnetcore/23.facebook-events/Bots/FacebookBot.cs [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+        private static async Task ShowChoices(ITurnContext turnContext, CancellationToken cancellationToken)
+        {
+            // Create choices for the prompt
+            var choices = new List<Choice>();
+            choices.Add(new Choice() { Value = QuickRepliesOption, Action = new CardAction() { Title = QuickRepliesOption, Type = ActionTypes.PostBack, Value = QuickRepliesOption } });
+            choices.Add(new Choice() { Value = FacebookPageIdOption, Action = new CardAction() { Title = FacebookPageIdOption, Type = ActionTypes.PostBack, Value = FacebookPageIdOption } });
+            choices.Add(new Choice() { Value = PostBackOption, Action = new CardAction() { Title = PostBackOption, Type = ActionTypes.PostBack, Value = PostBackOption } });
+
+            // Create the prompt message
+            var message = ChoiceFactory.ForChannel(turnContext.Activity.ChannelId, choices,"What Facebook feature would you like to try? Here are some quick replies to choose from!");
```
Comment (stevengum) [inline samples/csharp_dotnetcore/23.facebook-events/Bots/FacebookBot.cs]: Missing space: `choices,"What Facebook `

Comment (EricDahlvang) [reply samples/csharp_dotnetcore/23.facebook-events/Bots/FacebookBot.cs]: space added

**Pull request merged**